doc/notes/munin1.4.0.adoc

```diff
@@ -0,0 +1,192 @@
+= Munin 1.4.0 Release Notes
+
+== 1 Summary
+
+The Munin Protocol Verifier gained features and stability through Phase 2
+Stage 5.  As the Protocol Verifier readies for production deployment, a
+focus was made on deployment use cases, namely logging, reporting and
+continuous operation.  Logging and Reporting were a key focus with
+entirely new architectural infrastructure added to provide functionality
+and configurability.  An operations dashboard was constructed, refined and
+then used during testing.  Operational use cases such as starting,
+stopping, restarting and reconfiguring were practiced.  As in previous
+stages, collaboration was high among target users, testers and development
+engineering.
+
+As compared to the 1.3.0 release, the 1.4.0 release exhibits a reduction
+in throughput on the independent test fixture.  This is being analysed.
+More is explained in the 'Testing' section below.
+
+=== 1.1 Highlights
+
+Here is a list of areas improved in this release cycle.  More detail
+follows.
+
+Logging::
+The information and collection methods of logging are made richer and more
+flexible.
+
+Reporting::
+The collection, filtering and presentation of failures, alarms, throughput
+and other statistics has been implemented.
+
+json2plus::
+We already had 'plus2json', now we go the other way and derive PLUS
+(PlantUML .puml) from (reverse engineered) JSON job definitions.
+
+Hashing of Invariants::
+Invariants are no longer persisted as cleartext but are hashed.
+
+Throttling::
+The Protocol Verifier can now regulate its throughput and avoid overrun
+conditions.
+
+Start/Stop/Start::
+The Protocol Verifier can now start, stop and start again, picking up
+from where it left off.
+
+Testing and Bug Fixing::
+Any time we test, bugs are identified and addressed.
+
+== 2 Improvements
+
+=== 2.1 Logging
+
+The logging infrastructure migrated from `Poco` to a more functional and
+more widely used framework called `log4cplus`.  The primary motivation
+centered around the need to log through `Kafka`.  However `log4cplus`
+brings a host of additional features and functionality.  Highly useful in
+the Protocol Verifier is the ability to enable and disable various logging
+channels through configuration:
+
+* Logging can be configured to append to files, Kafka topics and/or syslogd.
+* The benchmark probe is now switchable.  It can be enabled and disabled
+  through configuration.
+* The PV Prime capability also can be switched on and off as needed
```

Comment:
Is it worth pointing out that the PVPrime instance of PV necessarily needs to be a **separate** instance rather than feeding the PV output back to itself - otherwise you get a runaway condition.